sys/mips/gt64120.c

```diff
@@ -246,27 +244,21 @@ static int gt_pci_attach(device_t *pcib) {
   gtpci->pci_mem =
     bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_PCI0_MEMORY_BASE,
                        MALTA_PCI0_MEMORY_END, MALTA_PCI0_MEMORY_SIZE, 0);
-  /* PCI I/O ports 0x1000-0xffff */
+  /* PCI I/O ports 0x0000-0xffff, mapped into KVA */
   gtpci->pci_io =
-    bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_PCI0_IO_BASE + 0x1000,
-                       MALTA_PCI0_IO_BASE + 0xffff, 0xf000, RF_ACTIVE);
-  /* GT64120 registers */
+    bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_PCI0_IO_BASE,
+                       MALTA_PCI0_IO_BASE + 0xffff, 0x10000, RF_ACTIVE);
+  /* GT64120 registers, mapped into KVA */
   gtpci->corectrl =
     bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_CORECTRL_BASE,
                        MALTA_CORECTRL_END, MALTA_CORECTRL_SIZE, RF_ACTIVE);
-  /* ISA I/O ports 0x0000-0x0fff */
-  gtpci->isa_io =
-    bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_PCI0_IO_BASE,
-                       MALTA_PCI0_IO_BASE + 0xfff, 0x1000, RF_ACTIVE);
 
   if (gtpci->corectrl == NULL || gtpci->pci_mem == NULL ||
-      gtpci->pci_io == NULL || gtpci->isa_io == NULL) {
+      gtpci->pci_io == NULL) {
     panic("gt64120 resource allocation fail");
   }
 
-  rman_init(&gtpci->isa_io_rman, "GT64120 ISA I/O ports", 0x0000, 0x0fff,
-            RT_IOPORTS);
-  rman_init(&gtpci->pci_io_rman, "GT64120 PCI I/O ports", 0x1000, 0xffff,
+  rman_init(&gtpci->pci_io_rman, "GT64120 PCI I/O ports", 0x0000, 0xffff,
             RT_IOPORTS);
   rman_init(&gtpci->pci_mem_rman, "GT64120 PCI memory", 0,
             MALTA_PCI0_MEMORY_SIZE, RT_MEMORY);
```

Comment:
Here should be `MALTA_PCI0_MEMORY_SIZE - 1`.

Reply:
Done.